app/models/kudo.rb

```diff
@@ -27,4 +29,19 @@ def name
   def dup?
     errors.values.to_s.match /already left kudos/
   end
+
+  def cannot_be_author
+    if self.pseud
+      commentable = self.commentable_type.classify.constantize.find_by_id(self.commentable_id)
+      kudos_giver = User.find_by_id(self.pseud.user_id)
+      if kudos_giver.is_author_of?(commentable)
+        errors.add(:cannot_be_author, ts("^You can't leave kudos on your own work."))
+      end
+    end
+  end
+  
+  def is_creator_of_work
```

Comment:
If you feel like indulging my preciousness, I would name the method `is_creator_of_work?`, for consistency with `dup?`.